<html><b>

Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show.

</b></html>

In [304]:
import pandas as pd
import numpy as np
import statistics


In [305]:
Brady_Bunch  = {'name': ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver'],
       'age': [14, 12, 11, 10, 8, 6, 8 ]}
labels = ['a', 'b', 'c', 'd', 'e', 'f', 'g']

df = pd.DataFrame(Brady_Bunch , index=labels)


In [306]:
print("\nThe Brady Bunch:")
print(df)


The Brady Bunch:
     name  age
a    Greg   14
b  Marcia   12
c   Peter   11
d     Jan   10
e   Bobby    8
f   Cindy    6
g  Oliver    8


<html><b>

What are the mean, median, and mode of the kids' ages when they first appeared on the show?

</b></html>

<html>
<b>
Mean
<b/>
</html>

In [307]:
# Using built-in Python functionality.
sum(df['age']) / len(df['age'])

9.857142857142858

In [308]:
# Using NumPy
np.mean(df['age'])

9.857142857142858

<html>
<b>
Median
<b/>
</html>

In [309]:
#Using Statistics Module
statistics.median(df['age'])

10

In [310]:
#Using Numpy
np.median(df['age'])

10.0

<html>
<b>
Mode
<b/>
</html>

In [311]:
#Using statistics module
statistics.mode(df['age'])

8

In [312]:
# (for study purposes)
# If two or more values in a variable occur with equal frequency, there will be 
# multiple modes
# Generate a list of unique elements along with how often they occur. 

(values, counts) = np.unique(df['age'], return_counts=True)

# The location in the values list of the most-frequently-occurring element.
ind = np.argmax(counts)

# The most frequent element.
values[ind]

8

<html><b>

What are the variance, standard deviation, and standard error?

</b></html>

<html><b>

Variance

</b></html>

In [313]:
#Using Pandas
#Use ddof=0 because the variance were differing. 
df['age'].var(ddof=0)

6.408163265306122

In [314]:
#Using Numpy
np.var(df.age)

6.408163265306122

<html><b>

Standard Deviation

</b></html>

<html>
    <i>
In standard statistical practice, ddof=1 provides an unbiased estimator of the variance of the infinite population. ddof=0 provides a maximum likelihood estimate of the variance for normally distributed variables. The standard deviation computed in this function is the square root of the estimated variance, so even with ddof=1, it will not be an unbiased estimate of the standard deviation per se.
    </i>    
</html>

In [315]:

np.std(df['age'], ddof=0)

2.531435020952764

<html><b>

Standard Error

</b></html>

In [316]:
np.std(df['age'] ,ddof=1) / np.sqrt(len(df['age']))

1.0334540197243192

<html>
<b>
Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?
</b>
</html>

I prefer numpy because in doing it with variance, it require simpler formulas and a better look at the answers in my opinion.

<html>
<b>
Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?
</b>
</html>

In [317]:
print("\nCindy has a birthday")
df.loc['f', 'age'] = 7
print(df)


Cindy has a birthday
     name  age
a    Greg   14
b  Marcia   12
c   Peter   11
d     Jan   10
e   Bobby    8
f   Cindy    7
g  Oliver    8


<html>
<b>
Mean
<b/>
</html>

In [318]:
# Using built-in Python functionality.
sum(df['age']) / len(df['age'])

10.0

In [319]:
# Using NumPy
np.mean(df['age'])

10.0

<html>
<b>
Mode
<b/>
</html>

In [320]:
#Using statistics module
statistics.mode(df['age'])

8

In [321]:
# (for study purposes)
# If two or more values in a variable occur with equal frequency, there will be 
# multiple modes
# Generate a list of unique elements along with how often they occur. 

(values, counts) = np.unique(df['age'], return_counts=True)

# The location in the values list of the most-frequently-occurring element.
ind = np.argmax(counts)

# The most frequent element.
values[ind]

8

<html><b>

Variance

</b></html>

In [322]:
#Using Pandas
#Use ddof=0 because the variance were differing. 
df['age'].var(ddof=0)

5.428571428571429

In [323]:
#Using Numpy
np.var(df.age)

5.428571428571429

<html>
<b></b>
    <b></b>
</html>

<html><b>

Standard Deviation

</b></html>

In [324]:
np.std(df['age'], ddof=1)

2.516611478423583

<html><b>

Standard Error

</b></html>

In [325]:
np.std(df['age'] ,ddof=1) / np.sqrt(len(df['age']))

0.9511897312113418

<html>
<b>Mean:</b> changed from 9.857142857142858 to 10.0.
    
<b>Median</b> remained the same.

<b>Mode:</b> remained the same.

<b>Variance:</b> changed from 6.408163265306122 to 5.428571428571429.

<b>Standard Deviation:</b> changed from 2.734262327610589 to 2.516611478423583.

<b>Standard Error:</b> changed from 1.0334540197243192 to 0.9511897312113418.

</html>

<html><b>Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods? </b><html>

In [326]:
print("\nGoodbye Oliver. Hello Jessica.")
df.loc['g', 'name']= 'Jessica'
df.loc['g', 'age'] = 1
print(df)


Goodbye Oliver. Hello Jessica.
      name  age
a     Greg   14
b   Marcia   12
c    Peter   11
d      Jan   10
e    Bobby    8
f    Cindy    7
g  Jessica    1


In [344]:
print("mean: ",np.mean(df.age))
print("median: ",np.median(df.age))
print("variance: ",np.var(df.age))
print("std. deviation :",np.std(df.age))
print("standard error :",np.std(df.age)/np.sqrt(len(df.age)-1))

print("There was no Mode")

#checked for mode with statistics module just in case. 
#Received error as expected.

mean:  9.0
median:  10.0
variance:  15.428571428571429
std. deviation : 3.927922024247863
standard error : 1.6035674514745466
There was no Mode


<html><b>Numpy is still prefered.</b></html>

<html><b>On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans

Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?.</b></html>

SciPhi, like Jessica, is an outlier and would change the stats too much so if I were to actually do a calculation to check percentages, it would not be included. Including the other three results in a 20% result for fans. 